### ISSUE TITLE
Difficult to step back into install process if you leave at "Install Theme" step

### ISSUE THREAD

**Issue by boonebgorges** [cuny-academic-commons/commons-in-a-box#144]
Copying from tests by @bree-z. @r-a-y When you have a chance, could you think about whether this part of the install process could be made more resilient?

> I've finished testing what happens if users don't strictly stay on task during the CBOX OL installation process.  It's pretty straightforward to get back on track, except once you get to the theme installation.  If you miss that, there's no way to get back to it, except using the back button, but that seemed to work fine.  And if you don't install the theme, you don't get the default settings for Groups, Members, Academic Units, etc.  Here's what I did:
> 
>     Selected CBOX OL and installed core required plugins 
>     Got to recommended plugins, which is where you see the CBOX OL section of the dashboard with all the sub-sections (Member Settings, Group Settings, etc).  
>     Instead of installing recommended plugins I started clicking around the various sub-sections of the dashboard.  It appears that you can do certain things at this point like add new member type or academic unit, (although there's no link to add group types at this point).  I also didn't actually try any of this, because I wanted to see if I could still complete the installation from here.  
>     I clicked on Dashboard, which took me back to my place in the installation process (i.e. install recommended plugins).
>     I installed recommended plugins, and then clicked 'Continue to theme installation.'
>     Here it notifies users that there's "One last step" and they see the 'Install Theme' button.  
>     At this point, if I clicked around in the Dashboard subsections, I saw the same as earlier, but clicking on the top-level Dashboard link didn't allow me to get back to the Install Theme page, like I could with plugin installation.  Clicking on that brought me to what looked like the dashboard of a completed installation, but it's not actually complete.  
>     I could only use the back button to get back to the theme installation page.  But, once I did this, I was able to successfully complete the installation.

**Comment by mkgold** [comment]
Hi Boone, Ray, and Bree,

Here’s to a more resilient CBOX! Boone, does this behavior occur only in CBOX-OL or also in CBOX Classic?

Best,

Matt
--
Matthew K. Gold, Ph.D
Associate Professor of English & Digital Humanities /
Advisor to the Provost for Digital Initiatives
The Graduate Center, CUNY
President, Association for Computers and the Humanities
http://cuny.is/mkgold

**Comment by boonebgorges** [comment]
CBOX-OL only, because that's the package where (a) the theme is required, and (b) the theme installation process kicks off the installation of a bunch of (required) default config data.

**Comment by r-a-y** [comment]
Good catch, @bree-z.  I didn't test for when a user would click elsewhere (such as another WordPress admin menu link) to exit the theme installation process.

As Boone mentioned above, if you do not install the theme, the CBOX-OpenLab-Core plugin does not trigger its set up process to add all its needed defaults.

The easiest thing to do would be to add some javascript to force a user decision on the Theme Installation step.  Either the user clicks on "Install Theme" or "Return to dashboard".  That way, we won't run into this scenario.

**Comment by bree-z** [comment]
Thanks, @r-a-y .  I may be misunderstanding, but I'm not sure that will solve the problem, because even if the user is at the Theme Installation step (after installing plugins), and clicks "Return to dashboard," and then wants to start the installation over again, they still won't be able to get back to the theme installation.  Here's what happens:

- User returns to dashboard and chooses CBOX OpenLab
- On OpenLab Details step, user chooses "Install"
- User gets a message that plugins are already installed.  
- After clicking button to continue, user ends up on the "Welcome to CBOX OpenLab" main dashboard page, but there's no way to get back to the theme installation step.

Does this make sense?  I could very well be missing something!  Thanks!

**Comment by r-a-y** [comment]
>  I may be misunderstanding, but I'm not sure that will solve the problem, because even if the user is at the Theme Installation step (after installing plugins), and clicks "Return to dashboard," and then wants to start the installation over again, they still won't be able to get back to the theme installation. 

The "Return to dashboard" link, in this instance, also resets the package marker.  So it's like starting the package installation process from scratch, meaning the admin should be able to return to the Theme Installation page that way.

We wanted to force the installation of the CBOX-OL theme, so that's why the "Return to dashboard" link functions this way.  Here's the relevant commit - https://github.com/cuny-academic-commons/commons-in-a-box/commit/15ee30a73e84569ab1530492af2e581216168ffa.

We could change the text to something else if we want it to be clearer.  Something like "Restart installation process" or "Choose another package".

**Comment by bree-z** [comment]
Thanks @r-a-y.  I'm guessing from your Basecamp message that clicking the Return to dashboard link doesn't reset the package marker as intended?

I also think changing the text to something more clear than "Return to dashboard" is a good idea.  I'm CC'ing @charliemedwards since we're talking about text changes.  I think "Restart installation process" makes sense, but she may have some thoughts.

Thanks!

**Comment by charliemedwards** [comment]
Hi there, it looks like people are in favor of "Return to package selection" (which was Ray's original suggestion on Basecamp, way back when).

**Comment by r-a-y** [comment]
I've narrowed down Bree's report.  I missed a scenario where if the recommended plugins were already active, that the theme installation step would be skipped.  This should now be fixed in commit dd51a9f.

Secondly, the scenario I mention where a user could click on another WP admin menu link to bypass the theme installation process is now addressed as of commit 8b20fe4.  The theme installation step is now always shown before the CBOX set up process is finalized.

Great to nail down these bugs!  Thanks again for the extensive testing, @bree-z.

@boonebgorges - Can you pull the latest commits so Bree can test?

**Comment by boonebgorges** [comment]
Thanks, @r-a-y - changes have been pulled to the hardg.com testing site.

**Comment by bree-z** [comment]
Thanks, all!  

I've tested again, and was able to get back to the theme installation step if (1) I deviated from the process by clicking on any of the other links in the dashboard menu and (2) if I clicked the "Return to dashboard" link.  

I guess this link still need to change to "Return to package selection" but perhaps that's part of the rest of the inline text changes Charlie and Matt are working on?

Thanks!
